File: server/src/main/java/com/fluffy/comment/application/dto/CreateExamCommentResponse.java
package com.fluffy.comment.application.dto;

import com.fluffy.auth.domain.Member;
import com.fluffy.comment.domain.ExamComment;
import java.time.LocalDateTime;

public record CreateExamCommentResponse(
        Long id,
        String content,
        Long examId,
        Long parentCommentId,
        AuthorResponse author,
        LocalDateTime createdAt
) {

    public static CreateExamCommentResponse of(ExamComment examComment, Member member) {
        return new CreateExamCommentResponse(
                examComment.getId(),
                examComment.getContent(),
                examComment.getExamId(),
                examComment.getParentCommentId(),
                AuthorResponse.from(member),
                examComment.getCreatedAt()
        );
    }
}

File: server/src/main/kotlin/com/fluffy/comment/application/response/CreateExamCommentResponse.kt
package com.fluffy.comment.application.response

import com.fluffy.auth.domain.Member
import com.fluffy.comment.domain.ExamComment
import java.time.LocalDateTime

data class CreateExamCommentResponse(
    val id: Long,
    val content: String,
    val examId: Long,
    val parentCommentId: Long?,
    val author: AuthorResponse,
    val createdAt: LocalDateTime,
) {

    companion object {

        fun of(examComment: ExamComment, member: Member): CreateExamCommentResponse {
            return CreateExamCommentResponse(
                examComment.id,
                examComment.content,
                examComment.examId,
                examComment.parentCommentId,
                AuthorResponse.from(member),
                examComment.createdAt,
            )
        }
    }
}
